src/panels/lovelace/cards/energy/hui-energy-usage-graph-card.ts

```diff
@@ -598,15 +650,16 @@ export class HuiEnergyUsageGraphCard
 
     Object.entries(combinedData).forEach(([type, sources]) => {
       Object.entries(sources).forEach(([statId, source], idx) => {
+        const overrideColor = colors[type].overrides?.[idx];
```

Comment:
```suggestion
        const overrideColor = computedStyles
            .getPropertyValue(colorPropertyMap[type] + "-" + idx)
            .trim();
```

Reply:
Instead of removing the color, we can just move the index a value, so use `colorIndex` for fetching the style, and bump that for every source?

Reply:
Oh right, I see your issue now

Reply:
Let me know if you'd still like to see some change here, now that we're on the same page.

Reply:
If we are going to iterate, let's then store the colors by `statId` so we don't have to do the `splice`

Reply:
Updated as requested.